src/main/g8/build.sbt

```diff
@@ -14,10 +14,33 @@ lazy val root = (project in file("."))
     // Options used by `native-image` when building native image.
     // https://www.graalvm.org/docs/reference-manual/native-image/
     graalVMNativeImageOptions ++= Seq(
-        "--initialize-at-build-time", // Auto-packs dependent libs at build-time
-        "--no-fallback", // Bakes-in run-time reflection (alternately: --auto-fallback, --force-fallback)
-        "--no-server", // Won't be running `graalvm-native-image:packageBin` often, so one less thing to break
-        "--static" // Disable for OSX (non-docker) builds - Forces statically-linked binary, compatible with libc (linux)
+      "--initialize-at-build-time", // Auto-packs dependent libs at build-time
+      "--no-fallback", // Bakes-in run-time reflection (alternately: --auto-fallback, --force-fallback)
+      "--no-server" // Won't be running `graalvm-native-image:packageBin` often, so one less thing to break
+    ),
+    // Additional options used by `native-image` when building native image that vary based on the
+    // operating system of the machine that is building the image
+    graalVMNativeImageOptions ++= {
+      // Note: Ideally, we should be checking if the current system has a working libc implementation here,
+      // but os.name works surprisingly well in most cases.
+      System.getProperty("os.name").toLowerCase match {
+        case mac if mac.contains("mac") =>
+          Nil
+        case win if win.contains("win") =>
+          Nil
+        case _ =>
+          // Forces statically-linked binary, compatible with libc (linux)
+          Seq("--static")
+      }
+    },
```

Comment:
This is definitely one of the more dubious parts of this PR. In order to make it so folks don't use this template and then IMMEDIATELY run into the issue of trying to build the image with the `--static` flag on their native macbook pro laptop, I added this code to do a little check. It's a bad check, for reasons explained in the comments, but it does make this thing a lot more "plug and play" for folks who are less inclined to RTFM.

Reply:
:+1: super fair. I think at one point early on I started to use these lines at my work, but threw it away after just a few weeks because I started to see folks treating it like voodoo magic.

TL;DR: I've come to believe that Graal native just plain ain't for folks who don't like RTFMing, so I totally agree with you that we shouldn't make gambles/sacrifices just to give them a 20% chance at things working out of the box (and 80% chance of frustrating failure with confusing error messages like the potential `NullPointerException` you mentioned)

Reply:
Interesting line of thought :thinking: 

Is the motivation for shoving all the things in a big ole' multi-project build to make maintaining the templates less difficult? Or is it that you actually DO wanna have your cake and 20 different kinds of ~plates to eat it on~ ways to run it too? Both? Neither?

Reply:
NGL, the idea of having one big multi-project that I can selectively delete things is appealing from a maintenance perspective.  

But for the use-case thing, I'll give some examples:

_Use-case 1:_  **Lambda functions on AWS**
I like to deploy lots of little jars, each of which encapsulate some domain logic for an endpoint or a binding of some sort like SQS or S3, and then version them separately.  They get connected via a consistent common/lib subproject behind them.

_Use-case 2:_ **CLI tool**
I like to make a "common" or "lib" subproject that I package separately, then bring into the actual application.  This is mainly to keep my dependencies separate and my interfaces clear so that I can reuse stuff that I slap into common with little effort.

Reply:
Only argument against doing it is this:

I'm thinking of someone linking this project to their friend who maybe just asked something along the lines of "I really want to play around with Graal Native, but I don't know where to start". As it stands now, this project is like aimed JUST broadly enough, yet also specifically enough at someone who wants to get started with GraalVM with sbt. 

If we start throwing other stuff in the mix, does that "dillute" the graaly parts too much? Four sub-projects sure doesn't sound too bad to me, and I DO really like the idea of dog-piling because I get the sense there is at least a solid chance of being able to pull it off without adding too much complexity. 

TL;DR: If you think that, after that multi-module update, this project can be something that folks can link as a sort of "first hit is free" kind of lure to get folks into Graal Native, then I'm all in for this change :+1: otherwise, well, I'll be nervous, but of course, I'm still in :laughing: